### PR TITLE
improvement: Bundling extension

### PR DESCRIPTION
**Pull request by 928PJY**
Bundling is the process of combining multiple small source files into a single file, loading 100 small files is much slower than loading one large file.

# Before bundling
Install package size: **5248KB**
Extension activation time: **520ms**
image

# After bundling
Install package size: **411KB**
Extension activation time: **74ms**
image

### CODE REVIEW

File: src/credential/credentialController.ts [hunk 1]
```diff
@@ -127,9 +128,9 @@ export class CredentialController {
     }
 
     private async signInWithAAD(): Promise<string> {
-        const authConfig = extensionConfig.auth[this.environmentController.env.toString()];
+        const authConfig = extensionConfig.auth[this.environmentController.env];
         const callbackUri = await vscode.env.asExternalUri(vscode.Uri.parse(`${vscode.env.uriScheme}://${EXTENSION_ID}/aad-authenticate`));
-        const signUrlTemplate = template.parse(`${AzureEnvironment.Azure.activeDirectoryEndpointUrl}/{tenantId}/oauth2/authorize` +
+        const signUrlTemplate = template.parse(`${AzureEnvironment.Azure.activeDirectoryEndpointUrl}{tenantId}/oauth2/authorize` +
```
Comment (fenxu) [inline src/credential/credentialController.ts]: >{AzureEnvironment.Azure.activeDirectoryEndpointUrl} [](start = 49, length = 51)

To keep safe, we always use TrimEnd('/') and then add '/'. That would make the url always correct without external dependency

File: .vscodeignore [hunk 2]
```diff
@@ -15,7 +15,11 @@ coverage/
 **/*.map
 **/*.ts
 **/*.gif
```
Comment (fenxu) [inline .vscodeignore]: why do we need to ignore all gifs? Will that cause problem?

Comment (928PJY) [reply .vscodeignore]: this file indicates the items which should be ignored when packaging the extension, anything which is not required in the runtime should be ignored to make the install package as small as possible.

Comment (fenxu) [reply .vscodeignore]: Can we use sth. like:

```
# Ignore everything
*
# Whitelist what you need
!src/**
```

Not sure whether that's possible, but just an idea.

---
In reply to: [373910436](https://github.com/docascode/vscode-docs-build/pull/32#discussion_r373910436) [](ancestors = 373910436)

File: src/credential/credentialController.ts (hunk 1 above)
Comment (herohua) [reply src/credential/credentialController.ts]: +1

---
In reply to: [373908512](https://github.com/docascode/vscode-docs-build/pull/32#discussion_r373908512) [](ancestors = 373908512)

**Review by herohua — APPROVED** [review]
:shipit:

File: .vscodeignore (hunk 2 above)
Comment (928PJY) [reply .vscodeignore]: that is a good idea, I will try, don't want to add almost everything in this file

**Review by fenxu — APPROVED** [review]
:shipit:

File: .vscodeignore (hunk 2 above)
Comment (928PJY) [reply .vscodeignore]: That works, good idea

**Pull request merged**